x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_edit.test.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as React from 'react';
+import { mountWithIntl } from 'test_utils/enzyme_helpers';
+import { act } from 'react-dom/test-utils';
+import { coreMock } from '../../../../../../../src/core/public/mocks';
+import { actionTypeRegistryMock } from '../../action_type_registry.mock';
+import { ValidationResult } from '../../../types';
+import { AppDeps } from '../../app';
+import { AlertsContextProvider } from '../../context/alerts_context';
+import { alertTypeRegistryMock } from '../../alert_type_registry.mock';
+import { chartPluginMock } from '../../../../../../../src/plugins/charts/public/mocks';
+import { dataPluginMock } from '../../../../../../../src/plugins/data/public/mocks';
+import { ReactWrapper } from 'enzyme';
+import { AlertEdit } from './alert_edit';
+const actionTypeRegistry = actionTypeRegistryMock.create();
+const alertTypeRegistry = alertTypeRegistryMock.create();
+
+describe('alert_edit', () => {
+  let deps: AppDeps | null;
+  let wrapper: ReactWrapper<any>;
+
+  beforeAll(async () => {
+    const mockes = coreMock.createSetup();
+    const [
+      {
+        chrome,
+        docLinks,
+        application: { capabilities },
+      },
+    ] = await mockes.getStartServices();
+    deps = {
+      chrome,
+      docLinks,
+      toastNotifications: mockes.notifications.toasts,
+      injectedMetadata: mockes.injectedMetadata,
+      http: mockes.http,
+      uiSettings: mockes.uiSettings,
+      dataPlugin: dataPluginMock.createStartContract(),
+      charts: chartPluginMock.createStartContract(),
+      capabilities: {
+        ...capabilities,
+        alerting: {
+          delete: true,
+          save: true,
+          show: true,
+        },
+      },
+      setBreadcrumbs: jest.fn(),
+      actionTypeRegistry: actionTypeRegistry as any,
+      alertTypeRegistry: alertTypeRegistry as any,
+    };
```

Comment:
I'm assuming we don't need *all* of these deps in this test, it's just because we're mocking the useDependencies?
It might be worth extracting this into separate reusable test files, like our mocks.